src/generic/AbsSeries.jl

```diff
@@ -23,7 +23,6 @@ function O(a::AbstractAlgebra.AbsSeriesElem{T}) where T <: RingElement
       return deepcopy(a)    # 0 + O(x^n)
    end
    prec = length(a) - 1
-   prec < 0 && throw(DomainError())
```

Comment:
`!iszero(a) && prec < 0` means `length(a) < 0`, which is an invalid state for `a`. I should fix that separately for series constructor.